engine/src/main/java/org/terasology/world/block/BlockRegion.java

```diff
@@ -0,0 +1,243 @@
+/*
+ * Copyright 2020 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.world.block;
+
+import org.joml.AABBi;
+import org.joml.Math;
+import org.joml.Matrix4fc;
+import org.joml.Rectanglei;
+import org.joml.RoundingMode;
+import org.joml.Vector3i;
+import org.joml.Vector3ic;
+import org.terasology.entitySystem.entity.EntityRef;
+
+public class BlockRegion extends AABBi {
```

Comment:
I don't like that this class extends `AABBi` and yet overrides every method to add a simple cast.
If it's done like that, you could as well not extend AABBi and have `BlockRegion` use an `AABBi` object internally instead.
This would also make the `BlockRegion` interface independent from `AABBi`.